### PR TITLE
Update perplexity, inherited from Metric

### PR DESCRIPTION
**Pull request by LiuChiachi**
Suitable for common machine translation task and language model.
已经在seq2seq和rnnlm下验证，与原ppl输出结果对齐

### CODE REVIEW

File: PaddleNLP/paddlenlp/metrics/perplexity.py [hunk 1]
```diff
@@ -15,33 +15,59 @@
...
+    """
+
+    def __init__(self, seq_len=20, name='Perplexity', *args, **kwargs):
         super(Perplexity, self).__init__(*args, **kwargs)
         self._name = name
-        self.total_ce = 0
-        self.num_batch = 0
+        self.total_cost = 0
+        self.word_count = 0
+        self.word_num = seq_len
```
Comment (smallv0221) [inline PaddleNLP/paddlenlp/metrics/perplexity.py]: 这个参数想办法在update中找到。

File: PaddleNLP/paddlenlp/metrics/perplexity.py [hunk 2]
```diff
@@ -15,33 +15,59 @@
...
+        name(str): Name of `Metric` instance. Default: 'Perplexity'.
+
+    """
+
+    def __init__(self, seq_len=20, name='Perplexity', *args, **kwargs):
         super(Perplexity, self).__init__(*args, **kwargs)
         self._name = name
-        self.total_ce = 0
-        self.num_batch = 0
+        self.total_cost = 0
```
Comment (smallv0221) [inline PaddleNLP/paddlenlp/metrics/perplexity.py]: ppl是基于cross-entropy的，cost都改成ce比较直观

File: PaddleNLP/paddlenlp/metrics/perplexity.py [hunk 3]
```diff
@@ -15,33 +15,59 @@
...
-    def update(self, y, label, *args):
-        # Perplexity is calculated using cross entropy
-        label = paddle.to_tensor(label)
-        y = paddle.to_tensor(y)
-        label = paddle.unsqueeze(label, axis=2)
+    def compute(self, pred, label, seq_mask=None):
+        cost = F.softmax_with_cross_entropy(
+            logits=pred, label=label, soft_label=False)
+        cost = paddle.squeeze(cost, axis=[2])
+        word_num = self.word_num
```
Comment (smallv0221) [inline PaddleNLP/paddlenlp/metrics/perplexity.py]: 这里的累加在语言模型中结果不对。

**Review by smallv0221 — CHANGES_REQUESTED** [review]

File: PaddleNLP/paddlenlp/metrics/perplexity.py (hunk 3 above)
Comment (LiuChiachi) [reply PaddleNLP/paddlenlp/metrics/perplexity.py]: 感谢，已经修改并在语言模型下验证了

File: PaddleNLP/paddlenlp/metrics/perplexity.py (hunk 2 above)
Comment (LiuChiachi) [reply PaddleNLP/paddlenlp/metrics/perplexity.py]: Thanks! Done.

File: PaddleNLP/paddlenlp/metrics/perplexity.py (hunk 1 above)
Comment (LiuChiachi) [reply PaddleNLP/paddlenlp/metrics/perplexity.py]: 感谢，已经修改~

File: PaddleNLP/paddlenlp/metrics/perplexity.py [hunk 4]
```diff
@@ -15,33 +15,61 @@
...
-        ce = paddle.mean(ce)
-
-        self.total_ce += ce.numpy()[0]
-        self.num_batch += 1
+    def update(self, ce, word_num=None):
+        batch_ce = np.sum(ce)
+        if word_num is None:
+            word_num = ce.shape[0] * ce.shape[1]
+        self.total_ce += batch_ce
+        self.word_count += word_num
```
Comment (smallv0221) [inline PaddleNLP/paddlenlp/metrics/perplexity.py]: 感觉word_count和word_num有点混淆，是不是第一个改成self.total_word好一点

Comment (LiuChiachi) [reply PaddleNLP/paddlenlp/metrics/perplexity.py]: Done. Thanks!

**Review by ZeyuChen — APPROVED** [review]
LGTM
